File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var dominantIndex = function (nums) {
    let fMax = -1;
    let sMax = -1;
    let ans = -1;

    for (let i = 0; i < nums.length; i++) {
        let val = nums[i];

        if (fMax < val) {
            sMax = fMax;
            fMax = val;
            ans = i
        } else if (sMax < val) {
            sMax = val;
        }

    }

    if (fMax >= sMax * 2) {
        return ans;
    } else {
        return -1;
    }

    return -1;
};

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.c
int dominantIndex(int* nums, int numsSize) {
    int f_Max = -1;
    int s_Max = -1;
    int ans = 0;

    for (int i = 0; i < numsSize; i++) {
        int val = nums[i];

        if (f_Max < val) {
            s_Max = f_Max;
            f_Max = val;
            ans = i;
        } else if (s_Max < val) {
            s_Max = val;
        }
    }

    if (f_Max >= s_Max * 2) {
        return ans;
    }

    return -1;
}